utils/hct/hctdb_instrhelp.py

```diff
@@ -537,35 +532,47 @@ def print_opfunc_props(self):
             "nr": "NoReturn",
             "wv": "None",
         }
-        attr_fn = lambda i: "Attribute::" + attr_dict[i.fn_attr] + ","
+        attr_fn = lambda i: "Attribute::" + attr_dict[i.fn_attr]
+        oload_to_mask = lambda oload: sum(
+            [1 << dxil_all_user_oload_chars.find(c) for c in oload]
+        )
+        oloads_fn = lambda oloads: (
+            "{" + ",".join(["{0x%x}" % m for m in oloads]) + "}"
+        )
         for i in self.instrs:
             if last_category != i.category:
                 if last_category != None:
                     print("")
-                print(
-                    "  // {category:118}  void,     h,     f,     d,    i1,    i8,   i16,   i32,   i64,   udt,   obj ,  function attribute".format(
-                        category=i.category
-                    )
-                )
+                if not i.is_reserved:
+                    print(f"  // {i.category}")
                 last_category = i.category
+            scalar_masks = []
+            vector_masks = []
+            if i.num_oloads > 0:
+                for n, o in enumerate(i.oload_types.split(",")):
+                    if "<" in o:
+                        v = o.split("<")
+                        scalar_masks.append(oload_to_mask(v[0]))
+                        vector_masks.append(oload_to_mask(v[1]))
+                    else:
+                        scalar_masks.append(oload_to_mask(o))
+                        vector_masks.append(0)
```

Comment:
I'm afraid this doesn't resolve the issue I ran into. Without "<" showing up in the "scalar" mask, the Vector bit doesn't get set in the allowedoverloads mask so IsOverloadLegal in DxilOperations.cpp rejects any vector oveload because it is a vector before it has a chance to check if the vector element is allowed.